contracts/contracts/vault/OETHVaultCore.sol

```diff
@@ -1,26 +1,147 @@
+    function cacheWETHAssetIndex() external onlyGovernor {
+        uint256 assetCount = allAssets.length;
+        for (uint256 i = 0; i < assetCount; ++i) {
+            if (allAssets[i] == weth) {
+                wethAssetIndex = i;
+                break;
+            }
+        }
+
+        require(allAssets[wethAssetIndex] == weth, "Invalid WETH Asset Index");
+    }
+
     // @inheritdoc VaultCore
-    function mint(
+    function _mint(
         address _asset,
         uint256 _amount,
         uint256 _minimumOusdAmount
-    ) external virtual override whenNotCapitalPaused nonReentrant {
+    ) internal virtual override {
         require(_asset == weth, "Unsupported asset for minting");
-        _mint(_asset, _amount, _minimumOusdAmount);
+        require(_amount > 0, "Amount must be greater than 0");
+        require(
+            _amount >= _minimumOusdAmount,
+            "Mint amount lower than minimum"
+        );
+
+        emit Mint(msg.sender, _amount);
+
+        // Rebase must happen before any transfers occur.
+        if (!rebasePaused && _amount >= rebaseThreshold) {
+            _rebase();
+        }
+
+        // Mint oTokens
+        oUSD.mint(msg.sender, _amount);
+
+        // Transfer the deposited coins to the vault
+        IERC20(_asset).safeTransferFrom(msg.sender, address(this), _amount);
+
+        // Auto-allocate if necessary
+        if (_amount >= autoAllocateThreshold) {
+            _allocate();
+        }
+    }
+
+    // @inheritdoc VaultCore
+    function _calculateRedeemOutputs(uint256 _amount)
+        internal
+        view
+        virtual
+        override
+        returns (uint256[] memory outputs)
+    {
+        // Overrides `VaultCore._calculateRedeemOutputs` to redeem with only
+        // WETH instead of LST-mix. Doesn't change the function signature
+        // for backward compatibility
+
+        // Calculate redeem fee
+        if (redeemFeeBps > 0) {
+            uint256 redeemFee = _amount.mulTruncateScale(redeemFeeBps, 1e4);
+            _amount = _amount - redeemFee;
+        }
+
+        // Ensure that the WETH index is cached
+        uint256 wethIndex = wethAssetIndex;
+        require(allAssets[wethIndex] == weth, "WETH Asset index not cached");
+
+        outputs = new uint256[](allAssets.length);
+        outputs[wethIndex] = _amount;
+    }
+
+    // @inheritdoc VaultCore
+    function _redeem(uint256 _amount, uint256 _minimumUnitAmount)
+        internal
+        virtual
+        override
+    {
+        // Override `VaultCore._redeem` to simplify it. Gets rid of oracle
+        // usage and looping through all assets for LST-mix redeem. Instead
+        // does a simple WETH-only redeem.
+        emit Redeem(msg.sender, _amount);
+
+        if (_amount == 0) {
+            return;
+        }
+
+        // Amount excluding fees
+        uint256 amountMinusFee = _calculateRedeemOutputs(_amount)[
+            wethAssetIndex
+        ];
+
+        if (_minimumUnitAmount > 0) {
```

Comment:
We can remove this if statement. It's now just an extra gas costs, since the computation inside costs the same as it does in the worst case, and in the normal case we have two if's getting called. A zero _minimumUnitAmount should still pass all checks.